File: 프로그래머스/2/42584. 주식가격/주식가격.py
def solution(prices):
    answer = [0 for _ in range(len(prices))]
    stack = []
    for idx in range(len(prices)):
        price= prices[idx]
        count = 0
        while len(stack) > 0 and stack[-1]["price"] > price:
            last = stack.pop()
            answer[last["idx"]] = idx - last["idx"] 
        stack.append({"idx": idx, "price": price})
    while len(stack) > 0:
        last = stack.pop()
        answer[last["idx"]] = len(prices) - last["idx"] - 1
    
    return answer

File: 프로그래머스/2/42584. 주식가격/주식가격.java
import java.util.*;
class Solution {
    public int[] solution(int[] prices) {
        int n = prices.length;
        Stack<int[]> stack = new Stack<>();
        int[] answer = new int[n];
        
        for(int i = 0; i< n ; i++){
            int price = prices[i];
            
            int count = 0;
            while(!stack.isEmpty() && stack.peek()[0] > price){
                int[] current = stack.pop();
                answer[current[1]] = i - current[1]; 
            }
            stack.push(new int[]{price, i});
        }
        
        while(!stack.isEmpty()){
            int[] current = stack.pop();
            answer[current[1]] = n - current[1] - 1; 
        }
        
        return answer;
    }
}